packages/features/bookings/components/event-meta/Duration.tsx

```diff
@@ -1,37 +1,65 @@
+import { useRouter } from "next/router";
 import { useEffect } from "react";
 
 import { useBookerStore } from "@calcom/features/bookings/Booker/store";
 import classNames from "@calcom/lib/classNames";
+import defaultEvents from "@calcom/lib/defaultEvents";
 import { useLocale } from "@calcom/lib/hooks/useLocale";
 import { Badge } from "@calcom/ui";
 
 import type { PublicEvent } from "../../types";
 
 export const EventDuration = ({ event }: { event: PublicEvent }) => {
   const { t } = useLocale();
+  const router = useRouter();
   const [selectedDuration, setSelectedDuration] = useBookerStore((state) => [
     state.selectedDuration,
     state.setSelectedDuration,
   ]);
 
+  const isDynamicEvent = "isDynamic" in event && event.isDynamic;
+
   // Sets initial value of selected duration to the default duration.
   useEffect(() => {
     // Only store event duration in url if event has multiple durations.
-    if (!selectedDuration && event.metadata?.multipleDuration) setSelectedDuration(event.length);
-  }, [selectedDuration, setSelectedDuration, event.length, event.metadata?.multipleDuration]);
+    if (!selectedDuration && (event.metadata?.multipleDuration || isDynamicEvent))
+      setSelectedDuration(event.length);
+  }, [selectedDuration, setSelectedDuration, event.metadata?.multipleDuration, event.length, isDynamicEvent]);
 
-  if (!event?.metadata?.multipleDuration) return <>{t("multiple_duration_mins", { count: event.length })}</>;
+  if (!event?.metadata?.multipleDuration && !isDynamicEvent)
+    return <>{t("multiple_duration_mins", { count: event.length })}</>;
 
   return (
     <div className="flex flex-wrap gap-2">
-      {event.metadata.multipleDuration.map((duration) => (
-        <Badge
-          variant="gray"
-          className={classNames(selectedDuration === duration && "bg-brand-default text-brand")}
-          size="md"
-          key={duration}
-          onClick={() => setSelectedDuration(duration)}>{`${duration} ${t("minute_timeUnit")}`}</Badge>
-      ))}
+      {event.metadata?.multipleDuration &&
```

Comment:
`Boolean(event.metaData?.,multipleDuration)` and `isDynamicEvent` can never be true at the same time, so we won't render the same badge again from line 43.
// see `packages/trpc/server/routers/publicViewer/event.handler.ts`